stwcs/wcsutil/headerlet.py

```diff
@@ -1969,9 +1970,12 @@ def apply_as_primary(self, fobj, attach=True, archive=True, force=False):
                                                                           suffix='_idc')])
                         else:
                             priwcs_name = 'UNKNOWN'
-                nextkey = altwcs.next_wcskey(fobj, ext=target_ext)
-                altwcs.archiveWCS(fobj, ext=sciext_list, wcskey=nextkey,
-                                  wcsname=priwcs_name)
+                    archive_wcs = True
```

Comment:
I wonder if we should compare `priwcs_name` and `wcsname` of the headerlet  before setting `archive_wcs=True`. This would be the case when the headerlet was created from the WCS.

Reply:
I think it is a great idea. One concern is what if the primary WCS was modified by the user (or some other function) but `WCSNAME` was not modified. Maybe also add a check to compare WCSes and primary WCS is the same as the one being applied from the headerlet - do not archive it. What is your opinion?